x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -116,16 +125,50 @@ export const CreateAnalyticsForm: FC<CreateAnalyticsFormProps> = ({ actions, sta
     }
   };
 
-  // Regression supports numeric fields. Classification supports numeric, boolean, text, keyword and ip.
-  const shouldAddFieldOption = (field: Field) => {
+  // Regression and Classification support numeric, boolean, text, keyword, and ip types
+  // Outlier detection supports numeric or boolean types
+  const shouldAddAsAnalyzedFieldsOption = (field: Field) => {
     if (field.id === EVENT_RATE_FIELD_ID) return false;
 
     const isNumerical = NUMERICAL_FIELD_TYPES.has(field.type);
-    const isSupportedByClassification =
-      isNumerical || SUPPORTED_CLASSIFICATION_FIELD_TYPES.has(field.type);
+
+    if (jobType === JOB_TYPES.OUTLIER_DETECTION)
+      return isNumerical || field.type === ES_FIELD_TYPES.BOOLEAN;
+    if (jobType === JOB_TYPES.REGRESSION || jobType === JOB_TYPES.CLASSIFICATION)
+      return isNumerical || SUPPORTED_ANALYSIS_FIELD_TYPES.has(field.type);
+  };
+
+  // Regression supports numeric fields. Classification supports numeric and boolean.
+  const shouldAddAsDepVarOption = (field: Field) => {
```

Comment:
Might be worth moving this logic out of the component file to some dedicated service, alongside with `shouldAddAsAnalyzedFieldsOption` and related variables like `SUPPORTED_CLASSIFICATION_FIELD_TYPES`

Reply:
Good call as we'll likely add more to this eventually. Moved to a separate file in 43d5ce5d93f16a7f28c5554d3f5db1eb0e580aac